docs/tutorials/edc-cut-bell-pair-benchmarking.ipynb

```diff
@@ -0,0 +1,715 @@
+    "    graph = defaultdict(set)\n",
+    "    for u, v in edges:\n",
+    "        graph[u].add(v)\n",
+    "        graph[v].add(u)\n",
+    "\n",
+    "    qubits = sorted(graph)  # all qubit indices that appear\n",
+    "\n",
+    "    # greedy search for 4-long linear chains (blocks) ────────────\n",
+    "    used = set()  # qubits already placed in a block\n",
+    "    blocks = []  # each block is a 4-qubit list\n",
+    "\n",
+    "    for q in qubits:  # deterministic order for reproducibility\n",
+    "        if q in used:\n",
+    "            continue  # already consumed by earlier block\n",
+    "\n",
+    "        # depth-first \"straight\" walk of length 3 without revisiting nodes\n",
+    "        def extend(path):\n",
+    "            if len(path) == 4:\n",
+    "                return path\n",
+    "            tip = path[-1]\n",
+    "            for nbr in sorted(graph[tip]):  # deterministic\n",
+    "                if nbr not in path and nbr not in used:\n",
+    "                    maybe = extend(path + [nbr])\n",
+    "                    if maybe:\n",
+    "                        return maybe\n",
+    "            return None\n",
+    "\n",
+    "        block = extend([q])\n",
+    "        if block:  # found a 4-node path\n",
+    "            blocks.append(block)\n",
+    "            used.update(block)\n",
+    "\n",
+    "    # bundle four 4-qubit blocks into one 16-qubit stripe (max number of measurement compatible with if-else)\n",
+    "    stripes = [\n",
+    "        list(chain.from_iterable(blocks[i : i + 4]))\n",
+    "        for i in range(0, len(blocks) // 4 * 4, 4)  # full groups of four\n",
+    "    ]\n",
+    "\n",
+    "    leftovers = set(qubits) - set(chain.from_iterable(stripes))\n",
+    "    return stripes, leftovers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 80,
+   "id": "3cf9db43",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "initial_layouts, leftover = stripes16_from_backend(backend)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8121aa97",
+   "metadata": {},
+   "source": [
+    "Next, we construct the circuit for each 16-qubit stripe. The routine does the following for each chain:\n",
+    "\n",
+    "* Prepare a middle Bell pair: Apply a Hadamard on qubit 1 and a CNOT from qubit 1 to qubit 2. This entangles qubits 1 and 2 (creating an $|\\Phi^+\\rangle = (|00\\rangle + |11\\rangle)/\\sqrt{2}$ Bell state).\n",
+    "* Entangle to the edge qubits: Apply a CNOT from qubit 0 to qubit 1, and a CNOT from qubit 2 to qubit 3. This links the initially separate pairs so that qubits 0 and 3 will become entangled after the next steps. A Hadamard on qubit 2 is also applied (this, combined with the previous CNOTs, forms part of a Bell measurement on qubits 1 and 2). At this point, qubits 0 and 3 are not yet entangled, but qubits 1 and 2 are entangled with them in a larger four-qubit state.\n",
```

Comment:
```suggestion
    "* Entangle the edge qubits: Apply a CNOT from qubit 0 to qubit 1, and a CNOT from qubit 2 to qubit 3. This links the initially separate pairs so that qubits 0 and 3 will become entangled after the next steps. A Hadamard on qubit 2 is also applied (this, combined with the previous CNOTs, forms part of a Bell measurement on qubits 1 and 2). At this point, qubits 0 and 3 are not yet entangled, but qubits 1 and 2 are entangled with them in a larger four-qubit state.\n",
```